### ISSUE TITLE
Support for Nest 2FA?

### ISSUE THREAD

**Issue by jabruder** [KhaosT/homebridge-nest-cam#16]
Hello,
First, thank you for making this plugin for Homebridge available. I was curious if support for the OAUTH / 2FA authentication that is in homebridge-nest might be brought into this project?

**Comment by ELLIOTTCABLE** [comment]
Yeah, I'm seeing:

```
[2018-2-10 08:04:33] [Nest-cam] Initializing Nest-cam platform...
[NestCam]Failed to request access token. The username and password do not match
```

Presumably, this requires a `"username"` and `"password"`, instead of the `"token"` from the other library. Bit of a bummer. 😞

**Comment by smartpuppy** [comment]
I would also like to see token support added like Homebridge-nest supports. Thanks.

**Comment by potmat** [comment]
I don't think this can happen, stepping through the code it looks like this plugin depends on a deprecated API that existed when these were still Dropcams. Only the camera v1 API can tell you how to pull video from a nest cam, the new APIs do not.  The only documentation for this API I could find is in the patent filing:

https://patents.google.com/patent/US9788039

I'm kind of surprised Google hasn't killed this off yet actually.

**Comment by KhaosT** [comment]
This is the API the app is using I believe.

**Comment by ChrisChiera** [comment]
Just to clarify this thread, does this plugin not support fully secured Nest accounts (ie 2 factor)? And based on the API limitations the plugin would not be possible when using more modern/secure API?

Was thrilled to see Nest Cam is now supported since love Nest and Nest Protect integration with Homebridge and Cam's was the one omission though of course wouldn't be able to disable 2FA since then anyone who guessed/found the login would have full access to our home making it probably the most important account to have 2FA on other than maybe email. Thanks!
